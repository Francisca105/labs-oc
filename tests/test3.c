#include "../code/4.2/L2Cache.h"

#define offset_bits 6
#define index_bits 8

// Checks if L2 Cache is used whenever possible instead of main memory

int main() {

  resetTime();
  initCache();

  int i = 0, value;
  write(i, (unsigned char *)(&i));
  printf("Write; Address %d; Value %d; Time %d\n", i, i, getTime());

  read(i, (unsigned char *)(&value));
  printf("Read; Address %d; Value %d; Time %d\n", i, value, getTime());

  int t1 = DRAM_READ_TIME + L2_READ_TIME + L1_WRITE_TIME + L1_READ_TIME;
  printf("Time = DRAM_READ_TIME + L2_READ_TIME + L1_WRITE_TIME + L1_READ_TIME = %d\n\n", t1);

  i = 1 << (offset_bits + index_bits);
  write(i, (unsigned char *)(&i));
  printf("Write; Address %d; Value %d; Time %d\n", i, i, getTime());

  read(i, (unsigned char *)(&value));
  printf("Read; Address %d; Value %d; Time %d\n", i, value, getTime());

  int t2 = t1 + DRAM_READ_TIME + L2_READ_TIME + L2_WRITE_TIME + L1_WRITE_TIME + L1_READ_TIME;
  printf("Time = Time + DRAM_READ_TIME + L2_READ_TIME + L2_WRITE_TIME + L1_WRITE_TIME + L1_READ_TIME = %d\n\n", t2);

  i = 0;
  write(i, (unsigned char *)(&i));
  printf("Write; Address %d; Value %d; Time %d\n", i, i, getTime());

  read(i, (unsigned char *)(&value));
  printf("Read; Address %d; Value %d; Time %d\n", i, value, getTime());

  int t3 = t2 + L2_READ_TIME + L2_WRITE_TIME + L1_WRITE_TIME + L1_READ_TIME;
  printf("Time = Time + L2_READ_TIME + L2_WRITE_TIME + L1_WRITE_TIME + L1_READ_TIME = %d\n", t3);
  
  return 0;
}