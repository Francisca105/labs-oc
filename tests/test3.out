Write; Address 0; Value 0; Time 111
Read; Address 0; Value 0; Time 112
Time = DRAM_READ_TIME + L2_READ_TIME + L1_WRITE_TIME + L1_READ_TIME = 112

Write; Address 16384; Value 16384; Time 228
Read; Address 16384; Value 16384; Time 229
Time = Time + DRAM_READ_TIME + L2_READ_TIME + L2_WRITE_TIME + L1_WRITE_TIME + L1_READ_TIME = 229

Write; Address 0; Value 0; Time 245
Read; Address 0; Value 0; Time 246
Time = Time + L2_READ_TIME + L2_WRITE_TIME + L1_WRITE_TIME + L1_READ_TIME = 246
